### PR TITLE
Configuring multiple helm chart repositories

### PR DESCRIPTION
**Pull request by Preeticp**
This PR covers the dev issue: https://issues.redhat.com/browse/APPSVC-385 and docs issue: https://issues.redhat.com/browse/RHDEVDOCS-2175
This is relevant to 4.6.

### CODE REVIEW

**Comment by Preeticp** [comment]
@pedjak @gajanan-more request you to review this PR, please.

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: modules/helm-creating-role-bindings-to-access-multiple-chart-repositories.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+// * cli_reference/helm/configuring-custom-helm-chart-repositories.adoc
+
+[id="creating-role-bindings-to-access-multiple-chart-repositories_{context}"]
+= Creating role bindings to access multiple chart repositories
+
+As a cluster administrator, you can add multiple instances of Helm chart repositories, and provide authenticated users access to charts from configured repositories by adding appropriate roles to the cluster.
+
+.Procedure
+
+* Provide regular users access to browse Helm chart repositories by creating the following role and role binding:
```
Comment (pedjak) [inline modules/helm-creating-role-bindings-to-access-multiple-chart-repositories.adoc:12]: this is already the default state in the cluster, hence there is no need to document it.

File: modules/helm-creating-role-bindings-to-access-multiple-chart-repositories.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  - kind: Group
+    apiGroup: rbac.authorization.k8s.io
+    name: 'system:authenticated'
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: azure-helm-repo-viewer
+----
+
+* For chart repositories where certificates are needed, provide users with read access to the `helm-ca-cert` configmap and `helm-tls-configs` secret in the `openshift-config` namespace:
```
Comment (pedjak) [inline modules/helm-creating-role-bindings-to-access-multiple-chart-repositories.adoc:81]: this needs to stay.

**Review by pedjak — COMMENTED** [review]
by default all authenticated users have access to all configured charts.

**Comment by Preeticp** [comment]
@pedjak I have removed the redundant information and moved the required role binding part to the Creating credentials and CA certificates to add Helm chart repositories section as a last step. Please take a look: https://helm-multi-rep--ocpdocs.netlify.app/openshift-enterprise/latest/cli_reference/helm_cli/configuring-custom-helm-chart-repositories.html

**Review by pedjak — APPROVED** [review]

**Comment by Preeticp** [comment]
@gajanan-more, PTAL, I have fixed the mention of Developer Perspective too.

File: cli_reference/helm_cli/configuring-custom-helm-chart-repositories.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[id="configuring-custom-helm-chart-repositories"]
+
+= Configuring custom Helm chart repositories
+include::modules/common-attributes.adoc[]
+:context: configuring-custom-helm-chart-repositories
+
+toc::[]
+
+The *Developer Catalog*, in the *Developer* perspective of the web console, displays the Helm charts available in the cluster. By default, it lists the Helm charts from the link:https://redhat-developer.github.io/redhat-helm-charts/index.yaml[Red Hat Helm chart repository]. As a cluster administrator, you can add multiple Helm chart repositories, apart from the default one, and display the Helm charts from these repositories in the *Developer Catalog*.
```
Comment (bmcelvee) [inline cli_reference/helm_cli/configuring-custom-helm-chart-repositories.adoc:9]: Is the link to the Red Hat Helm chart repository intended to open a file or direct to the actual repo?

File: modules/helm-adding-helm-chart-repositories.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/helm/configuring-custom-helm-chart-repositories.adoc
+
+[id="adding-helm-chart-repositories_{context}"]
+= Adding custom Helm chart repositories
+
```
Comment (bmcelvee) [inline modules/helm-adding-helm-chart-repositories.adoc:7]: Add short description of procedure (per the template)

File: modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+[id="creating-credentials-and-certificates-to-add-helm-repositories_{context}"]
+= Creating credentials and CA certificates to add Helm chart repositories
+
+Some Helm chart repositories need credentials and custom Certificate Authority (CA) certificates to connect to it. You can use the web console as well as the CLI to add credentials and certificates.
+
+.Procedure
+To configure the credentials and certificates, and then add a Helm chart repository using the CLI:
+
+. In the `openshift-config` namespace, create a `configmap` with a custom CA certificate in PEM encoded format, and store it under the `ca-bundle.crt` key within the configmap.
```
Comment (bmcelvee) [inline modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc:13]: s/configmap./configmap:

File: modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
++
+[source,terminal]
+----
+$ oc create configmap helm-ca-cert \
+--from-file=ca-bundle.crt=/path/to/certs/ca.crt \
+-n openshift-config
+----
++
+. In the `openshift-config` namespace, create a `secret`  to add the Client TLS Configurations. The client certificate and key must be in PEM encoded format and stored under the keys `tls.crt` and `tls.key`, respectively.
```
Comment (bmcelvee) [inline modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc:22]: Take this one or leave it :)

I see why you structured this one with the detail sentence following the action step (hence the take it or leave it) but for consistency, consider putting the command immediately following the action. E.g., 

. In the `openshift-config` namespace, create a `secret`  to add the Client TLS Configurations:
<command>
The client certificate and key must be in PEM encoded format and stored under the keys `tls.crt` and `tls.key`, respectively.

File: modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
++
+[source,terminal]
+----
+oc create secret generic helm-tls-configs \
```
Comment (bmcelvee) [inline modules/helm-creating-credentials-and-certificates-to-add-helm-repositories.adoc:26]: s/oc/$ oc

**Comment by bmcelvee** [comment]
Just a few questions/suggestions, otherwise LGTM! And just a reminder to squash commits before merging. :)

File: cli_reference/helm_cli/configuring-custom-helm-chart-repositories.adoc (hunk 3 above)
Comment (Preeticp) [reply cli_reference/helm_cli/configuring-custom-helm-chart-repositories.adoc:9]: Modified this a bit.

**Comment by Preeticp** [comment]
Thank you @bmcelvee I have fixed all the comments, PTAL

**Comment by bmcelvee** [comment]
LGTM - thank you!!

**Pull request merged**